mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/main_connexions/base/BaseReportSubmittedFragment.kt

```diff
@@ -0,0 +1,88 @@
+package rs.readahead.washington.mobile.views.fragment.main_connexions.base
+
+import android.os.Bundle
+import android.view.View
+import org.hzontal.shared_ui.bottomsheet.BottomSheetUtils
+import rs.readahead.washington.mobile.R
+import rs.readahead.washington.mobile.databinding.FragmentSendReportBinding
+import rs.readahead.washington.mobile.domain.entity.reports.ReportInstance
+import rs.readahead.washington.mobile.util.hide
+import rs.readahead.washington.mobile.views.base_ui.BaseBindingFragment
+import rs.readahead.washington.mobile.views.fragment.uwazi.widgets.ReportsFormEndView
+
+abstract class BaseReportSubmittedFragment :
+    BaseBindingFragment<FragmentSendReportBinding>(FragmentSendReportBinding::inflate) {
+
+    protected abstract val viewModel :BaseReportsViewModel
+    private lateinit var endView: ReportsFormEndView
+    private var reportInstance: ReportInstance? = null
+
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        initView()
+        initData()
+    }
+
+    private fun initData() {
+        with(viewModel) {
+            instanceDeleted.observe(viewLifecycleOwner) { title ->
+                title?.let {
+                    nav().popBackStack()
+                }
+
+            }
+        }
+    }
+
+    private fun initView() {
+        arguments?.let {
+            reportInstance = it.get(BUNDLE_REPORT_FORM_INSTANCE) as ReportInstance
+            showFormEndView()
+        }
+
+        binding.toolbar.backClickListener = {
+            nav().popBackStack()
+        }
+        binding.toolbar.onRightClickListener =
+            { reportInstance?.let { showDeleteBottomSheet(it) } }
+
+        binding.nextBtn.hide()
+
+    }
+
+    private fun showDeleteBottomSheet(entityInstance: ReportInstance) {
+        BottomSheetUtils.showConfirmSheet(
+            baseActivity.supportFragmentManager,
+            entityInstance.title,
+            getString(R.string.Delete_Report_Confirmation),
+            getString(R.string.action_yes),
+            getString(R.string.action_no), consumer = object : BottomSheetUtils.ActionConfirmed {
+                override fun accept(isConfirmed: Boolean) {
+                    if (isConfirmed) {
+                        viewModel.deleteReport(entityInstance)
+                    }
+                }
+            })
+    }
+
+    private fun showFormEndView() {
+        if (reportInstance == null) {
+            return
+        }
```

Comment:
Maybe we should call this method with passing the reportInstance if not null   reportInstance?.let { showFormEndView(it) }